### PR TITLE
Update local node runtime to version 14

### LINKED ISSUE #78: Is Node v12 too old?
`lsp_utils` installs Node v12 while some packages requires Node v14 like servers which comes from VSCode: `LSP-css`, `LSP-html` (not sure about the latest master json server in VSCode's repo).

The major problem using Node v14 is that if you are on Windows 7, you have to set env variable `NODE_SKIP_PLATFORM_CHECK` to `1` as it's not official supported. ([details](https://github.com/nodejs/node/issues/33000#issuecomment-644530517))

References
---

- Node.js release schedule: https://github.com/nodejs/Release#release-schedule
- Previous PR: https://github.com/sublimelsp/lsp_utils/pull/59

### PR DESCRIPTION
**Pull request by rchl**
For Windows 7 and lower set the NODE_SKIP_PLATFORM_CHECK=1 to allow it to work.

Fixes #78

### CODE REVIEW

**Comment by rchl** [comment]
I haven't tried it on Windows 7. Are you able to test it @jfcherng ?

**Comment by jfcherng** [comment]
I am getting this error msg:

```
LSP-json: The system Node.js does not meet the requirements: NodeRuntimePATH(node: None, npm: None, version: None)
Package Control: Skipping automatic upgrade, last run at 2021-09-02 10:32:49, next run at 2021-09-02 11:32:49 or after
[lsp_utils] Installing Node.js 14.17.6
LSP-json: Error: Ignoring local Node.js runtime due to an error. Error resolving node version:
Node.js is only supported on Windows 8.1, Windows Server 2012 R2, or higher.

Setting the NODE_SKIP_PLATFORM_CHECK environment variable to 1 skips this

check, but Node.js might not execute correctly. Any issues encountered on

unsupported platforms will not be fixed.
Unable to start subprocess for LSP-json
Traceback (most recent call last):
  File "C:\Users\Tester\Desktop\sublime_text_build_4113_x64\Data\Installed Packages\LSP.sublime-package\plugin/core/windows.py", line 335, in start_async
    if plugin_class.needs_update_or_installation():
  File "C:\Users\Tester\Desktop\SUBLIM~1\Data\Packages\LSP_UT~1\st3\lsp_utils\_client_handler\abstract_plugin.py", line 101, in needs_update_or_installation
    server = cls.get_server()
  File "C:\Users\Tester\Desktop\SUBLIM~1\Data\Packages\LSP_UT~1\st3\lsp_utils\npm_client_handler.py", line 92, in get_server
    'storage_path': cls.storage_path(),
  File "C:\Users\Tester\Desktop\SUBLIM~1\Data\Packages\LSP_UT~1\st3\lsp_utils\server_npm_resource.py", line 32, in create
    raise Exception('Failed resolving the Node Runtime. Please see Sublime Text console for more information')
Exception: Failed resolving the Node Runtime. Please see Sublime Text console for more information
```

File: st3/lsp_utils/node_runtime.py [hunk 1]
```diff
@@ -5,21 +5,24 @@
...
+import sys
 import tarfile
 import urllib.request
 import zipfile
 
 __all__ = ['NodeRuntime', 'NodeRuntimePATH', 'NodeRuntimeLocal']
 
 IS_MAC_ARM = sublime.platform() == 'osx' and sublime.arch() == 'arm64'
-DEFAULT_NODE_VERSION = '16.2.0' if IS_MAC_ARM else '12.20.2'
+IS_WINDOWS_7_OR_LOWER = sys.platform == 'win32' and sys.getwindowsversion() <= (6, 1)  # type: ignore
```
Comment (jfcherng) [inline st3/lsp_utils/node_runtime.py:23]:
```py
>>> sys.getwindowsversion() 
sys.getwindowsversion(major=6, minor=1, build=7601, platform=2, service_pack='Service Pack 1')
>>> sys.getwindowsversion() <= (6, 1)
False
>>> sys.getwindowsversion()[:2] <= (6, 1)
True
```

**Comment by jfcherng** [comment]
After fixing `IS_WINDOWS_7_OR_LOWER`, I have

```
Failed to run npm command "C:\Users\Tester\Desktop\sublime_text_build_4113_x64\Data\Package Storage\lsp_utils\node-runtime\14.17.6\node\node.exe C:\Users\Tester\Desktop\sublime_text_build_4113_x64\Data\Package Storage\lsp_utils\node-runtime\14.17.6\node\node_modules\npm\bin\npm-cli.js ci --scripts-prepend-node-path=true --verbose --production":
npm info it worked if it ends with ok
npm verb cli [
npm verb cli   'C:\\Users\\Tester\\Desktop\\sublime_text_build_4113_x64\\Data\\Package Storage\\lsp_utils\\node-runtime\\14.17.6\\node\\node.exe',
npm verb cli   'C:\\Users\\Tester\\Desktop\\sublime_text_build_4113_x64\\Data\\Package Storage\\lsp_utils\\node-runtime\\14.17.6\\node\\node_modules\\npm\\bin\\npm-cli.js',
npm verb cli   'ci',
npm verb cli   '--scripts-prepend-node-path=true',
npm verb cli   '--verbose',
npm verb cli   '--production'
npm verb cli ]
npm info using npm@6.14.15
npm info using node@v14.17.6
npm verb npm-session c4b1f25e2f161c85
npm info prepare initializing installer
npm verb prepare starting workers
npm verb prepare installation prefix: C:\Users\Tester\Desktop\sublime_text_build_4113_x64\Data\Package Storage\LSP-json\14.17.6\language-server
npm verb prepare using package-lock.json
npm verb checkLock verifying package-lock data
npm info prepare Done in 0.017s
npm verb extractTree extracting dependencies to node_modules/
npm verb teardown shutting down workers.
npm info teardown Done in 0s
npm verb type system
npm verb stack FetchError: request to https://registry.npmjs.org/vscode-json-languageserver/-/vscode-json-languageserver-1.3.4.tgz failed, reason: getaddrinfo EAI_FAIL registry.npmjs.org
npm verb stack     at ClientRequest.<anonymous> (C:\Users\Tester\Desktop\sublime_text_build_4113_x64\Data\Package Storage\lsp_utils\node-runtime\14.17.6\node\node_modules\npm\node_modules\node-fetch-npm\src\index.js:68:14)
npm verb stack     at ClientRequest.emit (events.js:400:28)
npm verb stack     at TLSSocket.socketErrorListener (_http_client.js:475:9)
npm verb stack     at TLSSocket.emit (events.js:400:28)
npm verb stack     at emitErrorNT (internal/streams/destroy.js:106:8)
npm verb stack     at emitErrorCloseNT (internal/streams/destroy.js:74:3)
npm verb stack     at processTicksAndRejections (internal/process/task_queues.js:82:21)
npm verb cwd C:\Users\Tester\Desktop\sublime_text_build_4113_x64\Data\Package Storage\LSP-json\14.17.6\language-server
npm verb Windows_NT 6.1.7601
npm verb argv "C:\\Users\\Tester\\Desktop\\sublime_text_build_4113_x64\\Data\\Package Storage\\lsp_utils\\node-runtime\\14.17.6\\node\\node.exe" "C:\\Users\\Tester\\Desktop\\sublime_text_build_4113_x64\\Data\\Package Storage\\lsp_utils\\node-runtime\\14.17.6\\node\\node_modules\\npm\\bin\\npm-cli.js" "ci" "--scripts-prepend-node-path=true" "--verbose" "--production"
npm verb node v14.17.6
npm verb npm  v6.14.15
npm ERR! code EAI_FAIL
npm ERR! errno EAI_FAIL
npm ERR! network request to https://registry.npmjs.org/vscode-json-languageserver/-/vscode-json-languageserver-1.3.4.tgz failed, reason: getaddrinfo EAI_FAIL registry.npmjs.org
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network 
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
npm verb exit [ 1, true ]
npm timing npm Completed in 1517ms

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Tester\npm-cache\_logs\2021-09-02T02_39_39_727Z-debug.log
```

**Comment by jfcherng** [comment]
Wait... even if I change it back to v12 `DEFAULT_NODE_VERSION = '16.2.0' if IS_MAC_ARM else '12.22.6'`, it installs v12, but installing the server gives the same error message.

**Comment by jfcherng** [comment]
I am not sure what goes wrong at this moment. Here are some tryings:

- If I use the latest release (i.e., not this PR) `lsp_utils` installed by PC and set `NODE_SKIP_PLATFORM_CHECK=1` manually in OS (i.e., not by codes). `lsp_utils` installs node `14.17.6` and `LSP-json` works without a problem.

File: st3/lsp_utils/node_runtime.py [hunk 2]
```diff
@@ -108,12 +111,17 @@ def meets_requirements(self) -> bool:
     def node_bin(self) -> Optional[str]:
         return self._node
 
+    def node_env(self) -> Optional[Dict[str, str]]:
+        if IS_WINDOWS_7_OR_LOWER:
+            return {'NODE_SKIP_PLATFORM_CHECK': '1'}
+        return None
```
Comment (jfcherng) [inline st3/lsp_utils/node_runtime.py:117]: It looks like we have to 

```py
    def node_env(self) -> Optional[Dict[str, str]]:
        env = os.environ
        if IS_WINDOWS_7_OR_LOWER:
            env.update({'NODE_SKIP_PLATFORM_CHECK': '1'})
        return env
```

Comment (rchl) [reply st3/lsp_utils/node_runtime.py:117]: Thanks. Did slightly different fix.

**Comment by jfcherng** [comment]
Apart from that weird test fail, I am fine with the current state.

**Review by jfcherng — APPROVED** [review]

**Pull request merged**